File: RepoTransBench/target_projects/C/Python/rxi_vec/tests/original/test_vec.py
import pytest
from src.vec import *

def intptrcmp(x):
    return x

def test_vec_push():
    v = Vec()
    vec_init(v)
    for i in range(1000):
        vec_push(v, i * 2)
    assert v.data[1] == 2
    assert v.data[999] == 999 * 2
    assert vec_push(v, 10) == 0
    vec_destroy(v)

def test_vec_pop():
    v = Vec()
    vec_init(v)
    vec_push(v, 123)
    vec_push(v, 456)
    vec_push(v, 789)
    assert vec_pop(v) == 789
    assert vec_pop(v) == 456
    assert vec_pop(v) == 123
    vec_destroy(v)

def test_vec_splice():
    v = Vec()
    vec_init(v)
    for i in range(1000):
        vec_push(v, i)
    vec_splice(v, 0, 10)
    assert v.data[0] == 10
    vec_splice(v, 10, 10)
    assert v.data[10] == 30
    vec_splice(v, v.length - 50, 50)
    assert v.data[v.length - 1] == 949
    vec_destroy(v)

def test_vec_swapsplice():
    v = Vec()
    vec_init(v)
    for i in range(10):
        vec_push(v, i)
    vec_swapsplice(v, 0, 3)
    assert v.data[0] == 7 and v.data[1] == 8 and v.data[2] == 9
    vec_swapsplice(v, v.length - 1, 1)
    assert v.data[v.length - 1] == 5
    vec_destroy(v)

def test_vec_insert():
    v = Vec()
    vec_init(v)
    for i in range(1000):
        vec_insert(v, 0, i)
    assert v.data[0] == 999
    assert v.data[v.length - 1] == 0
    vec_insert(v, 10, 123)
    assert v.data[10] == 123
    assert v.length == 1001
    vec_insert(v, v.length - 2, 678)
    assert v.data[999] == 678
    assert vec_insert(v, 10, 123) == 0
    vec_insert(v, v.length, 789)
    assert v.data[v.length - 1] == 789
    vec_destroy(v)

def test_vec_sort():
    v = Vec()
    vec_init(v)
    vec_push(v, 3)
    vec_push(v, -1)
    vec_push(v, 0)
    vec_sort(v)
    assert v.data[0] == -1
    assert v.data[1] == 0
    assert v.data[2] == 3
    vec_destroy(v)

def test_vec_swap():
    v = Vec()
    vec_init(v)
    vec_push(v, ord('a'))
    vec_push(v, ord('b'))
    vec_push(v, ord('c'))
    vec_swap(v, 0, 2)
    assert v.data[0] == ord('c') and v.data[2] == ord('a')
    vec_swap(v, 0, 1)
    assert v.data[0] == ord('b') and v.data[1] == ord('c')
    vec_swap(v, 1, 2)
    assert v.data[1] == ord('a') and v.data[2] == ord('c')
    vec_swap(v, 1, 1)
    assert v.data[1] == ord('a')
    vec_destroy(v)

def test_vec_truncate():
    v = Vec()
    vec_init(v)
    for i in range(1000):
        vec_push(v, 0)
    vec_truncate(v, 10000)
    assert v.length == 1000
    vec_truncate(v, 900)
    assert v.length == 900
    vec_destroy(v)

def test_vec_clear():
    v = Vec()
    vec_init(v)
    vec_push(v, 1)
    vec_push(v, 2)
    vec_clear(v)
    assert v.length == 0
    vec_destroy(v)

def test_vec_first():
    v = Vec()
    vec_init(v)
    vec_push(v, 0xf00d)
    vec_push(v, 0)
    assert vec_first(v) == 0xf00d
    vec_destroy(v)

def test_vec_last():
    v = Vec()
    vec_init(v)
    vec_push(v, 0)
    vec_push(v, 0xf00d)
    assert vec_last(v) == 0xf00d
    vec_destroy(v)

def test_vec_reserve():
    v = Vec()
    vec_init(v)
    vec_reserve(v, 100)
    assert v.capacity == 100
    vec_reserve(v, 50)
    assert v.capacity == 100
    vec_destroy(v)
    vec_init(v)
    vec_push(v, 123)
    vec_push(v, 456)
    vec_reserve(v, 200)
    assert v.capacity == 200
    assert vec_reserve(v, 300) == 0
    vec_destroy(v)

def test_vec_compact():
    v = Vec()
    vec_init(v)
    for i in range(1000):
        vec_push(v, 0)
    vec_truncate(v, 3)
    vec_compact(v)
    assert v.length == v.capacity
    assert vec_compact(v) == 0
    vec_destroy(v)

def test_vec_pusharr():
    a = [5, 6, 7, 8, 9]
    v = Vec()
    vec_init(v)
    vec_push(v, 1)
    vec_push(v, 2)
    vec_pusharr(v, a, 5)
    assert v.data[0] == 1
    assert v.data[2] == 5
    assert v.data[6] == 9
    vec_destroy(v)
    vec_init(v)
    vec_pusharr(v, a, 5)
    assert v.data[0] == 5
    vec_destroy(v)

def test_vec_extend():
    v = Vec()
    v2 = Vec()
    vec_init(v)
    vec_init(v2)
    vec_push(v, 12)
    vec_push(v, 34)
    vec_push(v2, 56)
    vec_push(v2, 78)
    vec_extend(v, v2)
    assert v.data[0] == 12 and v.data[1] == 34 and v.data[2] == 56 and v.data[3] == 78
    assert v.length == 4
    vec_destroy(v)
    vec_destroy(v2)

def test_vec_find():
    v = Vec()
    vec_init(v)
    for i in range(26):
        vec_push(v, ord('a') + i)
    assert vec_find(v, ord('a')) == 0
    assert vec_find(v, ord('z')) == 25
    assert vec_find(v, ord('d')) == 3
    assert vec_find(v, ord('_')) == -1
    vec_destroy(v)

def test_vec_remove():
    v = Vec()
    vec_init(v)
    for i in range(26):
        vec_push(v, ord('a') + i)
    vec_remove(v, ord('_'))
    assert v.length == 26
    vec_remove(v, ord('c'))
    assert v.data[0] == ord('a') and v.data[1] == ord('b') and v.data[2] == ord('d') and v.data[3] == ord('e')
    assert v.length == 25
    vec_destroy(v)

def test_vec_reverse():
    v = Vec()
    vec_init(v)
    vec_push(v, ord('a'))
    vec_push(v, ord('b'))
    vec_push(v, ord('c'))
    vec_push(v, ord('d'))
    vec_reverse(v)
    assert v.length == 4
    assert v.data[0] == ord('d') and v.data[1] == ord('c') and v.data[2] == ord('b') and v.data[3] == ord('a')
    vec_destroy(v)

def test_vec_foreach():
    v = Vec()
    vec_init(v)
    vec_push(v, 19)
    vec_push(v, 31)
    vec_push(v, 47)
    count = 0
    acc = 1
    for i, x in enumerate(v.data):
        acc *= (x + count)
        count += 1
    assert acc == (19 + 0) * (31 + 1) * (47 + 2)
    vec_destroy(v)

def test_vec_foreach_rev():
    v = Vec()
    vec_init(v)
    vec_push(v, 19)
    vec_push(v, 31)
    vec_push(v, 47)
    acc = 1
    count = 0
    for x in reversed(v.data):
        acc *= (x + count)
        count += 1
    assert acc == (19 + 2) * (31 + 1) * (47 + 0)
    vec_destroy(v)

def test_vec_foreach_ptr():
    v = Vec()
    vec_init(v)
    vec_push(v, 19)
    vec_push(v, 31)
    vec_push(v, 47)
    count = 0
    acc = 1
    for i, x in enumerate(v.data):
        acc *= (x + count)
        count += 1
    assert acc == (19 + 0) * (31 + 1) * (47 + 2)
    vec_destroy(v)

def test_vec_foreach_ptr_rev():
    v = Vec()
    vec_init(v)
    vec_push(v, 19)
    vec_push(v, 31)
    vec_push(v, 47)
    acc = 1
    count = 0
    for x in reversed(v.data):
        acc *= (x + count)
        count += 1
    assert acc == (19 + 2) * (31 + 1) * (47 + 0)
    vec_destroy(v)

File: RepoTransBench/target_projects/C/Rust/rxi_vec/tests/original/test_vec.rs
// Translated from test/test_vec.c
use rxi_vec::{VecInt, VecDouble};

fn intptrcmp(a: &i32, b: &i32) -> std::cmp::Ordering {
    a.cmp(b)
}

#[test]
fn test_vec_push() {
    let mut v = VecInt::new();
    for i in 0..1000 {
        v.push(i * 2);
    }
    assert_eq!(v.data[1], 2);
    assert_eq!(v.data[999], 999 * 2);
    assert_eq!(v.push(10), 0);
}

#[test]
fn test_vec_pop() {
    let mut v = VecInt::new();
    v.push(123);
    v.push(456);
    v.push(789);
    assert_eq!(v.pop(), 789);
    assert_eq!(v.pop(), 456);
    assert_eq!(v.pop(), 123);
}

#[test]
fn test_vec_splice() {
    let mut v = VecInt::new();
    for i in 0..1000 {
        v.push(i);
    }
    v.splice(0, 10);
    assert_eq!(v.data[0], 10);
    v.splice(10, 10);
    assert_eq!(v.data[10], 30);
    let len = v.len();
    v.splice(len - 50, 50);
    assert_eq!(v.data[v.len() - 1], 949);
}

#[test]
fn test_vec_swapsplice() {
    let mut v = VecInt::new();
    for i in 0..10 {
        v.push(i);
    }
    v.swapsplice(0, 3);
    assert_eq!(v.data[0], 7);
    assert_eq!(v.data[1], 8);
    assert_eq!(v.data[2], 9);
    v.swapsplice(v.len() - 1, 1);
    assert_eq!(v.data[v.len() - 1], 5);
}

#[test]
fn test_vec_insert() {
    let mut v = VecInt::new();
    for i in 0..1000 {
        v.insert(0, i);
    }
    assert_eq!(v.data[0], 999);
    assert_eq!(v.data[v.len() - 1], 0);
    v.insert(10, 123);
    assert_eq!(v.data[10], 123);
    assert_eq!(v.len(), 1001);
    v.insert(v.len() - 2, 678);
    assert_eq!(v.data[999], 678);
    assert_eq!(v.insert(10, 123), 0);
    v.insert(v.len(), 789);
    assert_eq!(v.data[v.len() - 1], 789);
}

#[test]
fn test_vec_sort() {
    let mut v = VecInt::new();
    v.push(3);
    v.push(-1);
    v.push(0);
    v.data.sort_by(intptrcmp);
    assert_eq!(v.data[0], -1);
    assert_eq!(v.data[1], 0);
    assert_eq!(v.data[2], 3);
}

#[test]
fn test_vec_swap() {
    let mut v = VecInt::new();
    v.push('a' as i32);
    v.push('b' as i32);
    v.push('c' as i32);
    v.swap(0, 2);
    assert_eq!(v.data[0], 'c' as i32);
    assert_eq!(v.data[2], 'a' as i32);
    v.swap(0, 1);
    assert_eq!(v.data[0], 'b' as i32);
    assert_eq!(v.data[1], 'c' as i32);
    v.swap(1, 2);
    assert_eq!(v.data[1], 'a' as i32);
    assert_eq!(v.data[2], 'c' as i32);
    v.swap(1, 1);
    assert_eq!(v.data[1], 'a' as i32);
}

#[test]
fn test_vec_truncate() {
    let mut v = VecInt::new();
    for _ in 0..1000 {
        v.push(0);
    }
    v.truncate(10000);
    assert_eq!(v.len(), 1000);
    v.truncate(900);
    assert_eq!(v.len(), 900);
}

#[test]
fn test_vec_clear() {
    let mut v = VecInt::new();
    v.push(1);
    v.push(2);
    v.clear();
    assert_eq!(v.len(), 0);
}

#[test]
fn test_vec_first() {
    let mut v = VecInt::new();
    v.push(0xf00d);
    v.push(0);
    assert_eq!(v.first(), 0xf00d);
}

#[test]
fn test_vec_last() {
    let mut v = VecInt::new();
    v.push(0);
    v.push(0xf00d);
    assert_eq!(v.last(), 0xf00d);
}

#[test]
fn test_vec_reserve() {
    let mut v = VecInt::new();
    v.reserve(100);
    assert!(v.capacity() >= 100);
    v.reserve(50);
    assert!(v.capacity() >= 100);
    let mut v2 = VecInt::new();
    v2.push(123);
    v2.push(456);
    v2.reserve(200);
    assert!(v2.capacity() >= 200);
    assert_eq!(v2.reserve(300), 0);
}

#[test]
fn test_vec_compact() {
    let mut v = VecInt::new();
    for _ in 0..1000 { v.push(0); }
    v.truncate(3);
    v.compact();
    assert_eq!(v.len(), v.capacity());
    assert_eq!(v.compact(), 0);
}

#[test]
fn test_vec_pusharr() {
    let a = [5, 6, 7, 8, 9];
    let mut v = VecDouble::new();
    v.push(1.0);
    v.push(2.0);
    v.pusharr(&a, 5);
    let d = &v.data;
    assert_eq!(d[0], 1.0);
    assert_eq!(d[2], 5.0);
    assert_eq!(d[6], 9.0);
    let mut v2 = VecDouble::new();
    v2.pusharr(&a, 5);
    assert_eq!(v2.data[0], 5.0);
}

#[test]
fn test_vec_extend() {
    let mut v = VecInt::new();
    let mut v2 = VecInt::new();
    v.push(12);
    v.push(34);
    v2.push(56);
    v2.push(78);
    v.extend(&v2);
    assert_eq!(v.data[0], 12);
    assert_eq!(v.data[1], 34);
    assert_eq!(v.data[2], 56);
    assert_eq!(v.data[3], 78);
    assert_eq!(v.len(), 4);
}

#[test]
fn test_vec_find() {
    let mut v = VecInt::new();
    for i in 0..26 {
        v.push('a' as i32 + i);
    }
    assert_eq!(v.find('a' as i32), 0);
    assert_eq!(v.find('z' as i32), 25);
    assert_eq!(v.find('d' as i32), 3);
    assert_eq!(v.find('_' as i32), -1);
}

#[test]
fn test_vec_remove() {
    let mut v = VecInt::new();
    for i in 0..26 {
        v.push('a' as i32 + i);
    }
    v.remove('_' as i32);
    assert_eq!(v.len(), 26);
    v.remove('c' as i32);
    assert_eq!(v.data[0], 'a' as i32);
    assert_eq!(v.data[1], 'b' as i32);
    assert_eq!(v.data[2], 'd' as i32);
    assert_eq!(v.data[3], 'e' as i32);
    assert_eq!(v.len(), 25);
}

#[test]
fn test_vec_reverse() {
    let mut v = VecInt::new();
    v.push('a' as i32);
    v.push('b' as i32);
    v.push('c' as i32);
    v.push('d' as i32);
    v.reverse();
    assert_eq!(v.len(), 4);
    assert_eq!(v.data[0], 'd' as i32);
    assert_eq!(v.data[1], 'c' as i32);
    assert_eq!(v.data[2], 'b' as i32);
    assert_eq!(v.data[3], 'a' as i32);
}

#[test]
fn test_vec_foreach() {
    let mut v = VecInt::new();
    v.push(19); v.push(31); v.push(47);
    let mut count = 0;
    let mut acc = 1;
    for (i, &x) in v.data.iter().enumerate() {
        acc *= x + count;
        count += 1;
    }
    assert_eq!(acc, (19 + 0) * (31 + 1) * (47 + 2));
}

#[test]
fn test_vec_foreach_rev() {
    let mut v = VecInt::new();
    v.push(19); v.push(31); v.push(47);
    let mut xs = v.data.clone();
    xs.reverse();
    let mut count = 0;
    let mut acc = 1;
    for (i, &x) in xs.iter().enumerate() {
        acc *= x + count;
        count += 1;
    }
    assert_eq!(acc, (19 + 2) * (31 + 1) * (47 + 0));
}

#[test]
fn test_vec_foreach_ptr() {
    let mut v = VecInt::new();
    v.push(19); v.push(31); v.push(47);
    let mut count = 0;
    let mut acc = 1;
    for (i, x) in v.data.iter().enumerate() {
        acc *= *x + count;
        count += 1;
    }
    assert_eq!(acc, (19 + 0) * (31 + 1) * (47 + 2));
}

#[test]
fn test_vec_foreach_ptr_rev() {
    let mut v = VecInt::new();
    v.push(19); v.push(31); v.push(47);
    let mut xs = v.data.clone();
    xs.reverse();
    let mut count = 0;
    let mut acc = 1;
    for (i, x) in xs.iter().enumerate() {
        acc *= *x + count;
        count += 1;
    }
    assert_eq!(acc, (19 + 2) * (31 + 1) * (47 + 0));
}